nltk/translate/bleu_score.py

```diff
@@ -140,10 +140,14 @@ def corpus_bleu(list_of_references, hypotheses, weights=(0.25, 0.25, 0.25, 0.25)
     :type weights: list(float)
     :param smoothing_function:
     :type smoothing_function: SmoothingFunction
-    :param auto_reweigh:
+    :param auto_reweigh: Option to re-normalize the weights uniformly.
     :type auto_reweigh: bool
-    :param emulate_multibleu: bool
+    :param emulate_multibleu: Option to emulate multi-bleu.pl behavior
```

Comment:
Can you add here. "If set to `True` the original equation proposed by K Papineni et. al". 

That is much clearer than "Option to emulate multi-bleu.pl behavior", which might be useful for you who have read through their code, but is not useful for external users who haven't

Reply:
Actually, it's not exactly the original equation proposed by Papineni et al. because on paper it didn't really account for the case where no. of 4-grams overlap is < 0. There's no resolution until mteval.pl was released openly.

But I understand the docstring for the parameter needs to be more obvious. I'll make the changes.

Reply:
I assume you mean "didn't really account for the case where no. of 4-grams overlap is < 0"?

In that case their paper did account for that case as well. If you look at the equation for p_n in section 2.1 you can see that if we have 0 4-grams overlap the numerator will evaluate to 0 which means that p_n = 0. And log(0) should evaluate to -inf in this case! If you don't believe me ask somebody else or email the authors for example. 

What they didn't specify is when the reference sentence is shorter than N, where N is the order of N-grams used. I.e. when we have a 0/0 or x/0 situation. This is the thing I intend to solve later so that we can let 0/0 evaluate to 1 in that case.

Reply:
I believe you with the -inf case =)

I meant to say:

 - 0/1 = 0  and math.log(0) = -inf is a python problem (math error)
 - 1/0 = undefined, maths error 
 - and that Papineni et al. never accounted for sentence BLEU 

How about just changing the parameter name to `original_bleu=True` by default?

Reply:
Also, see the non-inline comment. We can avoid all this is we make one of the smooothing method the default for sentence BLEU. 

The argument is that there is BLEU vs sentence BLEU and sentence BLEU should never exist as unsmoothed. Does that make sense?

Reply:
I think changing the parameter name to `original_bleu` would be good, and very explanatory for the user yes :)

What should be default is up to you. However if you ask me I think `original_bleu = True` as default makes sense. Typically when you start using some new software tool you expect the default parameters to be "bare-bones", without any extra stuff etc. By the same reason I also think that the current implementation of not using any smoothing function by default, but raising a warning to urge the user to do so is a nice approach.

And yes, exactly that is what I mean, when there's zero 4-gram overlaps Papini et. al. specifies what to do. But when there isn't even 4 tokens in the reference sentence (and thus 0 4-grams) they do not specify what to do. 

You're right, they never mention sentence BLEU. And as you say corpus BLEU is probably mostly what is of interest rather than sentence BLEU. We actually use sentence BLEU and average afterwards, I don't know if that's the best, but it works well.